### PR TITLE
Add Logstash Telemetry

### PR DESCRIPTION
**Pull request by robbavey**
This PR adds telemetry related to Logstash

### CODE REVIEW

**Comment by robbavey** [comment]
PR ready for review

File: pkg/telemetry/telemetry.go [hunk 1]
```diff
@@ -398,6 +405,33 @@ func agentStats(k8sClient k8s.Client, managedNamespaces []string) (string, inter
...
+		stackMonitoringMetricsCount: 0, serviceCount: 0, pipelinesCount: 0, pipelinesRefCount: 0}
+
+	var logstashList logstashv1alpha1.LogstashList
+	for _, ns := range managedNamespaces {
+		if err := k8sClient.List(context.Background(), &logstashList, client.InNamespace(ns)); err != nil {
+			return "", nil, err
+		}
+
+		for _, ls := range logstashList.Items {
+			ls := ls
```
Comment (kaisecheng) [inline pkg/telemetry/telemetry.go:419]: remove it?

Comment (robbavey) [reply pkg/telemetry/telemetry.go:419]: This one was caught by the linter, and is related to loop variable semantics in go - there is a great summary of the issue (and proposal for "fixing" it in future versions of golang) [here](https://github.com/golang/go/discussions/56010). I don't actually think that that behaviour will bite us here, given what we are doing, but I'm still happy keeping this "odd" looking assignment in here and keep the linter satisfied

**Comment by robbavey** [comment]
@pebrc @barkbay This is ready for review by the ECK team

File: pkg/telemetry/telemetry.go [hunk 2]
```diff
@@ -41,7 +42,12 @@ const (
 	podCount                 = "pod_count"
 	helmManagedResourceCount = "helm_resource_count"
 
-	timestampFieldName = "timestamp"
+	timestampFieldName          = "timestamp"
+	stackMonitoringLogsCount    = "stack_monitoring_logs_count"
+	stackMonitoringMetricsCount = "stack_monitoring_metrics_count"
+	serviceCount                = "service_count"
+	pipelinesCount              = "pipelines_count"
+	pipelinesRefCount           = "pipelines_ref_count"
```
Comment (thbkrkr) [inline pkg/telemetry/telemetry.go:50]:
```suggestion
	stackMonitoringLogsCount    = "stack_monitoring_logs_count"
	stackMonitoringMetricsCount = "stack_monitoring_metrics_count"
	serviceCount                = "service_count"
	pipelineCount              = "pipeline_count"
	pipelineRefCount           = "pipeline_ref_count"
```

No strong opinion on this but we started to use singular for almost all `xxx_count`.

Nit: Move these new vars just under `helmManagedResourceCount`.

To go further, I'd keep these constants local in the `logstashStats` func waiting to improve more globally the telemetry (check my other comment). 
It's not very clean at the moment, things are not reusable (this is historycal tech debt not from your PR) and for example now we end up with a duplicate `stack_monitoring_logs_count:

https://github.com/elastic/cloud-on-k8s/blob/beaa45767c2c22179f6e318705a5395bbf4a7779/pkg/telemetry/telemetry.go#L232

**Review by thbkrkr — COMMENTED** [review]
LGTM, left a few nits.

It's not from your PR, but I realize that we started having duplicates and making the stats slightly different from app to app, causing us to miss some stats (stack mon for Kibana, or helm for agent and beats).

I think we could improve that by creating some reusable structs like this.

```go
type baseStats struct {
	ResourceCount int32 `json:"resource_count"`
	PodCount      int32 `json:"pod_count"`
}

type helmStats struct {
	HelmManagedResourceCount int32 `json:"helm_resource_count"`
}
type stackMonStats struct {
	StackMonitoringLogsCount    int32 `json:"stack_monitoring_logs_count"`
	StackMonitoringMetricsCount int32 `json:"stack_monitoring_metrics_count"`
}

func logstashStats(k8sClient k8s.Client, managedNamespaces []string) (string, interface{}, error) {
	logstashStats := struct {
		baseStats
		helmStats
		stackMonStats
		ServiceCount     int32 `json:"service_count"`
		PipelineCount    int32 `json:"pipeline_count"`
		PipelineRefCount int32 `json:"pipeline_ref_count"`
	}{}
...
```

I made a sketch [here](https://github.com/thbkrkr/cloud-on-k8s/commit/07b62c0ca2de1df476449b87a710c562bc0a40ab).

It's not completely trivial so it can be done in another PR, unless you're motivated to do it in this one :).

File: pkg/telemetry/telemetry.go [hunk 3]
```diff
@@ -398,6 +405,36 @@ func agentStats(k8sClient k8s.Client, managedNamespaces []string) (string, inter
...
+		}
+
+		for _, ls := range logstashList.Items {
+			ls := ls
+			stats[resourceCount]++
+			stats[serviceCount] += int32(len(ls.Spec.Services))
+			stats[podCount] += ls.Status.AvailableNodes
+			stats[pipelinesCount] += int32(len(ls.Spec.Pipelines))
+			if ls.Spec.PipelinesRef != nil {
+				stats[pipelinesRefCount] ++
```
[the remark this answers is not in the document]
Comment (thbkrkr) [reply pkg/telemetry/telemetry.go:425]: This is what causes the build to fail.

**Comment by robbavey** [comment]
@thbkrkr Ready for another round

**Review by thbkrkr — APPROVED** [review]

**Pull request merged**